server/platform/shared/web/files.go

```diff
@@ -91,3 +93,53 @@ func WriteFileResponse(filename string, contentType string, contentSize int64, l
 
 	http.ServeContent(w, r, filename, lastModification, fileReader)
 }
+
+func WriteStreamResponse(w http.ResponseWriter, r io.ReadCloser, filename string, contentType string, forceDownload bool) error {
```

Comment:
This reminds me a lot of the above method. I am sure you thought about potentially consolidating into a single thing. Just noting it, no ask here.

Reply:
Honestly, I didn't want to touch core code if I didn't have to (a little too worried of introducing bugs). But you're right, it should be compressed. Done!